components/compliance-service/dao/pgdb/upgrades.go

```diff
@@ -32,29 +33,61 @@ func (u *UpgradesDB) UpdateControlFlagToFalse() error {
 	return nil
 }
 
-//GetControlLatestUpgradeFlag Gets the day latest flag status from the pg database
-func (u *UpgradesDB) GetControlLatestUpgradeFlagValue() (status bool, err error) {
-	logrus.Info("Inside the get day latest flag")
-	err = u.DB.QueryRow(getQueryForFlag(ControlIndexFlag)).Scan(&status)
+//UpdateCompRunInfoFlagToTrue updates the day latest flags to true
+func (u *UpgradesDB) UpdateCompRunInfoFlagToTrue() error {
+	_, err := u.DB.Exec(getUpdateQuery(CompRunInfoFlag))
 	if err != nil {
-		return status, errors.Wrapf(err, "PollForUpgradeFlagDayLatest unable to get current job status")
+		return errors.Wrapf(err, "Unable to UpdateDayLatestFlagToTrue")
 	}
+	return nil
+}
+
+//GetControlLatestUpgradeFlag Gets the day latest flag status from the pg database
+func (u *UpgradesDB) GetControlLatestUpgradeFlagValue() (status bool, err error) {
+	logrus.Info("Inside the get control index flag")
 	return status, err
 }
 
 //GetDayLatestUpgradeFlagValue Gets the day latest flag status from the pg database
 func (u *UpgradesDB) GetDayLatestUpgradeFlagValue() (status bool, err error) {
-	logrus.Info("Inside the get day latest flag")
-	err = u.DB.QueryRow(getQueryForFlag(DayLatestFlag)).Scan(&status)
+	logrus.Info("Inside the day latest flag")
+	return status, err
+}
+
+//GetDayLatestUpgradeFlagValue Gets the day latest flag status from the pg database
+func (u *UpgradesDB) GetUpgradeFlags() (map[string]bool, error) {
+	flagMap := make(map[string]bool)
+
+	logrus.Info("Inside the comp run info flag")
+	flags := []string{CompRunInfoFlag, ControlIndexFlag, DayLatestFlag}
+	rows, err := u.DB.Query(getQueryForFlag(flags))
 	if err != nil {
-		return status, errors.Wrapf(err, "PollForUpgradeFlagDayLatest unable to get current job status")
+		return flagMap, err
 	}
-	return status, err
+	defer func() {
+		if err := rows.Close(); err != nil {
+			logrus.Errorf("failed to close db rows: %s", err.Error())
+		}
+	}()
+
+	for rows.Next() {
```

Comment:
there is nothing wrong while writing this block, but I remember there is some other way using which we can get the slice from rows.